internal/metrics/metrics.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	// DeploymentGauge tracks the number of deployments seen by the controller
+	// Labels: namespace, can_create_pdb (true/false)
+	DeploymentGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eviction_autoscaler_deployments_total",
+			Help: "Total number of deployments seen by the eviction autoscaler",
+		},
+		[]string{"namespace", "can_create_pdb"},
+	)
+
+	// PDBGauge tracks the number of PDBs seen by the controller
+	// Labels: namespace, created_by_us (true/false)
+	PDBGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eviction_autoscaler_pdbs_total",
+			Help: "Total number of PDBs seen by the eviction autoscaler",
+		},
+		[]string{"namespace", "created_by_us"},
+	)
+
+	// EvictionCounter tracks how often the eviction-autoscaler notices an eviction
+	// Labels: namespace, pod_name
+	EvictionCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_evictions_total",
+			Help: "Total number of evictions noticed by the eviction autoscaler",
+		},
+		[]string{"namespace", "pod_name"},
+	)
+
+	// BlockedEvictionCounter tracks how often evictions are blocked by PDBs
+	// Labels: namespace, pdb_name, pod_name
+	BlockedEvictionCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "eviction_autoscaler_blocked_evictions_total",
+			Help: "Total number of evictions blocked by PDBs",
+		},
+		[]string{"namespace", "pdb_name", "pod_name"},
+	)
+
+	// ScalingOpportunityCounter tracks how often the controller thinks it could have scaled a deployment
+	// Labels: namespace, deployment_name, action (scale_up/scale_down)
+	ScalingOpportunityCounter = prometheus.NewCounterVec(
```

Comment:
Add two more labels to this 
OldNotReadyPods
MinAvailableEqualsReplicas
maxUnavailable

Going to try and use these as evidence that the scaling would have helped.

Reply:
OldNotReadyPods could also be a more generic LikelyToHelp that we can define later.

Reply:
Maybe just a generic signal label then we fill in what the signal is.